python/lancedb/table.py

```diff
@@ -130,6 +133,24 @@ def create_index(self, metric="L2", num_partitions=256, num_sub_vectors=96):
         )
         self._reset_dataset()
 
+    def create_fts_index(self, field_names: Union[str, List[str]]):
```

Comment:
can we share the `create_index()` API ?

Reply:
We can, it just makes that api harder to use because now you have to figure out the index type name and also which kwargs go with which index type etc.

Reply:
would that make a bunch of `create_foo_index()` ? i.e., a `create_btree_index()` or `create_vector_index()`

Reply:
I think we can refactor this into an Index/IndexBuilder abstraction. I just didn't want to commit to it now because this is an experimental feature.